### PR TITLE
Add docs on running live tests serially

### PR DESCRIPTION
**Pull request by heaths**
- General clean-up of markdownlint issues
- Add section on serial live tests

### CODE REVIEW

**Comment by heaths** [comment]
I included some general clean-up for markdownlint errors as recommended by the Microsoft Learn Extension Pack for Visual Studio Code (same rules as required for https://learn.microsoft.com, which makes spotting actual errors/warnings easier!).

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/core/Azure.Core.TestFramework/README.md [hunk 1]
```diff
@@ -147,31 +146,53 @@ public class AppConfigurationTestEnvironment : TestEnvironment
 }
 ```
 
+### Running live tests serially
+
+By default, NUnit does not run tests within each assembly in parallel, but this be [configured](https://docs.nunit.org/articles/nunit/technical-notes/usage/Framework-Parallel-Test-Execution.html).
+Especially for unit tests, this is often desirable; however, live and [recorded tests](#recorded-tests) may run into some issues. Thus, by default, the `RecordedTestBase` described below is attributed
+as `[NonParallelizable]`.
+
+However, when projects are built and tested in CIs, all projects are testing in parallel. This means, for example, you can have two or more assemblies running tests such as one backing up or restoring
```
Comment (jsquire) [inline sdk/core/Azure.Core.TestFramework/README.md:155]:
```suggestion
However, when projects are built and tested in CI pipelines, all projects are testing in parallel. This means, for example, you can have two or more assemblies running tests such as one backing up or restoring
```

**Review by jsquire — APPROVED** [review]

**Review by christothes — APPROVED** [review]

File: sdk/core/Azure.Core.TestFramework/README.md [hunk 2]
```diff
@@ -147,31 +146,53 @@ public class AppConfigurationTestEnvironment : TestEnvironment
 }
 ```
 
+### Running live tests serially
```
Comment (annelo-msft) [inline sdk/core/Azure.Core.TestFramework/README.md:149]: Thanks for adding this - I've been wanting to know how to do this for the ACR library!

**Review by pallavit — APPROVED** [review]

File: sdk/core/Azure.Core.TestFramework/README.md (hunk 2 above)
Comment (JoshLove-msft) [reply sdk/core/Azure.Core.TestFramework/README.md:149]: +1 - I'm going to add this for Service Bus which will hopefully address some of the flakiness in the Functions extensions tests.

File: sdk/core/Azure.Core.TestFramework/README.md (hunk 1 above)
Comment (heaths) [reply sdk/core/Azure.Core.TestFramework/README.md:155]: CIs - Continuous Integration - already imply some form of pipeline, but I don't see harm in taking this if it helps clear it up.

**Pull request merged**